core/src/main/java/com/radixdlt/consensus/sync/BFTSync.java

```diff
@@ -442,12 +455,20 @@ private void rebuildAndSyncQC(SyncState syncState) {
       syncState.fetched.sort(Comparator.comparing(v -> v.vertex().getRound()));
       ImmutableList<VertexWithHash> nonRootVertices =
           syncState.fetched.stream().skip(1).collect(ImmutableList.toImmutableList());
+
+      final var syncStateHighestCommittedQc = syncState.highQC().highestCommittedQC();
+      final var syncStateHighestTc = syncState.highQC.highestTC();
+      final var currentHighestTc = vertexStore.highQC().highestTC();
+      final var highestKnownTc =
+          Stream.of(currentHighestTc, syncStateHighestTc)
+              .flatMap(Optional::stream)
+              .max(Comparator.comparing(TimeoutCertificate::getRound));
+
       var vertexStoreState =
           VertexStoreState.create(
-              HighQC.from(syncState.highQC().highestCommittedQC()),
+              HighQC.from(syncStateHighestCommittedQc, syncStateHighestCommittedQc, highestKnownTc),
```

Comment:
Oh dang. Nice catch here.

TBH this method `HighQC.from(qc)` feels wrong - what do you think about removing this constructor for `HighQC` and make people be explicit about committed / timeout?

Reply:
Happy to have an explicit method `HighQC.ofInitialEpochQc` or something though

Reply:
:+1: renamed to `ofInitialEpochQc`

Reply:
(and added a genesis check)